### PR TITLE
Change Availability to include timing and contain LocalTime variables

### LINKED ISSUE #124: Alter Availability Feature to enforce start and end time in LocalTime, HH:mm format
- [ ] Amend Availability Feature #120 
- [ ] Update Test Data #120 
- [ ] Update UserGuide Examples #123

### PR DESCRIPTION
**Pull request by whitesnowx**
- [x] Amend test files 

_Will update UserGuide in a seperate PR_ --> #123 

Closes #124

### CODE REVIEW

File: src/main/java/staffconnect/model/availability/Availability.java [hunk 1]
```diff
@@ -4,18 +4,36 @@
...
+    public static final String MESSAGE_CONSTRAINTS = "[DAY_OF_THE_WEEK] [START_TIME] [END_TIME] E.g. mon 09:00 13:00\n"
+            + "[DAY_OF_THE_WEEK] Should be a week of the day, the full word or the first syllable of the word\n"
+            + "[START_TIME], [END_TIME] Using a 24H digital time format.\n";
+    public static final String MESSAGE_INVALID_TIME = "Invalid Time, Format: HH:mm\n"
+            + "End time must be after Start time";
+    public static final String MESSAGE_END_BEFORE_START = "Format: [DAY_OF_THE_WEEK] [START_TIME] [END_TIME]\n"
+            + "End time must be after Start time";
+    public static final String VALIDATION_REGEX_DAY = "(?i)((mon|tue(s)?|wed(nes)?|thu(r)?(rs)?|fri|sat(ur)?|sun)"
+            + "(day)?)";
+    public static final String VALIDATION_REGEX_TIME = "([01][0-9]|2[0-3]):([0-5][0-9])";
```
Comment (Pluiexo) [inline src/main/java/staffconnect/model/availability/Availability.java:27]: I think you missed out on this, I don't see this being used anywhere

File: src/main/java/staffconnect/model/availability/Availability.java [hunk 2]
```diff
@@ -4,18 +4,36 @@
...
+    public static final String VALIDATION_REGEX = VALIDATION_REGEX_DAY + " [0-9]{2}:[0-9]{2} [0-9]{2}:[0-9]{2}";
```
Comment (Pluiexo) [inline src/main/java/staffconnect/model/availability/Availability.java:23]: I think you missed out on this, I don't see ```VALIDATION_REGEX_TIME``` being used anywhere

File: src/main/java/staffconnect/model/availability/Availability.java [hunk 3]
```diff
@@ -25,40 +43,79 @@ public class Availability {
...
+        if (!test.matches(VALIDATION_REGEX)) {
+            return false;
+        }
+
+        String [] availabilityFields = test.split(" ");
+        DayOfWeek testDay = parseToDayOfWeek(availabilityFields[0]);
+
+        if (!isValidLocalTime(availabilityFields[1]) || !isValidLocalTime(availabilityFields[2])) {
+            return false;
+        }
```
Comment (Pluiexo) [inline src/main/java/staffconnect/model/availability/Availability.java:60]: I think its better to reorder this to the top, with your guard clauses on top and logic below

File: src/main/java/staffconnect/model/availability/Availability.java [hunk 4]
```diff
@@ -25,40 +43,79 @@ public class Availability {
...
         } else {
-            return DayOfWeek.SUNDAY.name();
+            return DayOfWeek.SUNDAY;
         }
     }
 
+    /**
+     * Returns a standardised LocalTime.
+     *
+     * @param validTime validTime to be translated.
```
Comment (Pluiexo) [inline src/main/java/staffconnect/model/availability/Availability.java]: repeated name validTime

**Review by Pluiexo — COMMENTED** [review]
Lgtm, though minor nits

File: src/main/java/staffconnect/model/availability/Availability.java (hunk 4 above)
Comment (whitesnowx) [reply src/main/java/staffconnect/model/availability/Availability.java]: _Was intentional but bad documentation_
Fixed: changed param description

File: src/main/java/staffconnect/model/availability/Availability.java (hunk 2 above)
Comment (whitesnowx) [reply src/main/java/staffconnect/model/availability/Availability.java:23]: Fixed: removed

File: src/main/java/staffconnect/model/availability/Availability.java (hunk 3 above)
Comment (whitesnowx) [reply src/main/java/staffconnect/model/availability/Availability.java:60]: Are you referring to reordering to be on top of `String [] availabilityFields = test.split(" ");`

Comment (Pluiexo) [reply src/main/java/staffconnect/model/availability/Availability.java:60]: Oh okay my bad I did not notice the ```test.split(" ")``` there, then i think it is okay to leave it as it is

**Review by Pluiexo — APPROVED** [review]
Lgtm

**Pull request merged**